scipy/stats/morestats.py

```diff
@@ -3478,3 +3478,76 @@ def circstd(samples, high=2*pi, low=0, axis=None, nan_policy='propagate'):
         R = np.minimum(1, hypot(sin_mean, cos_mean))
 
     return ((high - low)/2.0/pi) * sqrt(-2*log(R))
+
+
+def quartile_coeff_dispersion(a, q=(0.25, 0.75), axis=0,
+                              interpolation='lower'):
+    """
+    Compute the quartile coefficient of dispersion for a given sample array.
+
+    Given two quartiles Q_low, Q_high from array samples such that
+    Q_low < Q_high, the quartile coefficient of dispersion is calculated by
+    (Q_high - Q_low)/(Q_high + Q_low).
+    by default it is computed using 3rd and 1st quartiles.
+
+    Parameters
+    ----------
+    a : array_like
+        Input array.
+    q : array_like of float, optional
+        Sequence of 2 quantiles, such that q[0] < q[1].
+        The default is to compute with Q_high = 0.75, Q_low = 0.25.
+    axis : {int, tuple of int, None}, optional
+        Axis along which the coefficient of dispersion is computed.
+        Default is 0. If None, compute over the whole array a.
+    interpolation : {'linear', 'lower', 'higher', 'midpoint', 'nearest'}
+        This optional parameter specifies the interpolation method to
+        use when the desired quantile lies between two data points
+        ``i < j``:
+
+        * linear: ``i + (j - i) * fraction``, where ``fraction``
+          is the fractional part of the index surrounded by ``i``
+          and ``j``.
+        * lower: ``i``.
+        * higher: ``j``.
+        * nearest: ``i`` or ``j``, whichever is nearest.
+        * midpoint: ``(i + j) / 2``.
+
+        The default is 'lower'.
+
+    References
+    ----------
+    .. [1] "Quartile Coefficient of Dispersion", *Wikipedia*,
+    https://en.wikipedia.org/wiki/Quartile_coefficient_of_dispersion.
+
+    Returns
+    -------
+    dispersion : {float, array_like of float}
+        Dispersion for each axis slice.
+
+    Examples
+    --------
+    >>> from scipy.stats import quartile_coeff_dispersion
+    >>> quartile_coeff_dispersion(a=[2, 4, 6, 8, 10, 12, 14, 16])
+    0.5
+
+    >>> quartile_coeff_dispersion(a=[1.6, 2.1, 2.3, 2.4,
+    ...                           2.6, 2.9, 2.98, 3],
+    ...                           q=(0.25, 0.5),
+    ...                           interpolation='midpoint')
+    0.0638297872340425
+
+    """
+    if np.shape(q) != (2,):
+        raise ValueError("Shape of q must be (2,)")
+
+    if q[0] < 0 or q[1] > 1:
+        raise ValueError("q values must be 2 floats in range (0.0, 1.0)")
+
+    if q[0] >= q[1]:
+        raise ValueError("q values should satisfy q[0] < q[1]")
+
+    quartiles = np.quantile(a, q=(q[0], q[1]),
+                            axis=axis, interpolation=interpolation)
+
+    return (quartiles[1] - quartiles[0])/(quartiles[1] + quartiles[0])
```

Comment:
Question to other reviewer: are warnings on divide by zero OK? Result is nan, as I suppose it should be.